src/connectionbenchmark/connectionbenchmark.cpp

```diff
@@ -40,10 +40,20 @@ void ConnectionBenchmark::initialize() {
 void ConnectionBenchmark::setConnectionSpeed() {
   logger.debug() << "Set connection speed";
 
-  // TODO: Take uploadBps for calculating speed into account
-  if (m_downloadBps >= Constants::BENCHMARK_THRESHOLD_SPEED_FAST) {
+  bool uploadIsAvailable =
+      Feature::get(Feature::Feature_benchmarkUpload)->isSupported();
+
+  if (m_downloadBps >= Constants::BENCHMARK_THRESHOLD_DOWN_FAST &&
```

Comment:
You can simplify these if-stmts doing:
`if ... && (!upload is available || upload >= x)`

Reply:
Thanks for the suggestion! I removed the upload threshold again since we decided to only take the result of the download benchmark into account for setting the connection speed.